x-pack/plugins/reporting/server/browsers/chromium/driver/index.ts

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as Chrome from 'devtools-protocol';
+import fs from 'fs';
+import moment from 'moment';
+import path from 'path';
+import util from 'util';
+import { ElementPosition, Logger } from '../../../../types';
+import { ignoreSSLErrorsBehavior } from './ignore_ssl_errors';
+import { screenshotStitcher } from './screenshot_stitcher';
+import { transformFn } from './transform_fn';
+
+const WAIT_FOR_DELAY_MS: number = 100;
+
+export class HeadlessChromiumDriver {
+  private client: Chrome.Protocol.ProtocolApi;
+  private maxScreenshotDimension: number;
+  private zoom: number = 1;
+  private logger: Logger;
+  private killed: boolean = false;
+  private documentNode?: Chrome.Protocol.DOM.GetDocumentResponse;
+
+  constructor(
+    client: Chrome.Protocol.ProtocolApi,
+    options: { maxScreenshotDimension: number; logger: Logger }
+  ) {
+    this.client = client;
+    this.maxScreenshotDimension = options.maxScreenshotDimension;
+    this.logger = options.logger;
+  }
+
+  public async evaluate<T>({
+    fn,
+    args = [],
+    awaitPromise = false,
+    returnByValue = false,
+  }: {
+    fn: () => T;
+    args: Array<string | object>;
+    awaitPromise?: boolean;
+    returnByValue?: boolean;
+  }) {
+    const { Runtime } = this.client;
+
+    const serializedArgs = args.map(arg => JSON.stringify(arg)).join(',');
+    const expression = `(${transformFn(fn)})(${serializedArgs})`;
+    const result = await Runtime.evaluate({
+      awaitPromise,
+      expression,
+      returnByValue,
+    });
+    return result.result.value;
+  }
+
+  public async open(
+    url: string,
+    { headers, waitForSelector }: { headers: { [key: string]: string }; waitForSelector: string }
+  ) {
+    this.logger.debug(`HeadlessChromiumDriver:opening url ${url}`);
+    const { Network, Page } = this.client;
+    await Promise.all([Network.enable({}), Page.enable()]);
+
+    await ignoreSSLErrorsBehavior(this.client.Security);
+    await Network.setExtraHTTPHeaders({ headers });
+    await Page.navigate({ url });
+
+    await new Promise(resolve => Page.on('loadEventFired', resolve));
```

Comment:
Changed this around to fit with the types, though Page.loadEventFired *should* work per their docs, it's not whats in the type library (see https://github.com/ChromeDevTools/devtools-protocol/issues/104)